src/ethereum/prague/transactions.py

```diff
@@ -117,7 +117,7 @@ class SetCodeTransaction:
     The transaction type added in EIP-7702.
     """
 
-    chain_id: U64
+    chain_id: U256
```

Comment:
The EIP still says

```
The fields chain_id, nonce, max_priority_fee_per_gas, max_fee_per_gas, gas_limit, destination, value, data, and access_list of the outer transaction follow the same semantics as [EIP-4844](https://github.com/lightclient/EIPs/blob/7702-chainid-u256/EIPS/eip-4844.md).
```

I looked back at 4844, 1559 and a couple of other ones in the series and none of them clearly define whether `chain_id` is `U64` or `U256`. In fact, EIP-1559 has some pseudo-code that looks at it as `int`

So I suggest explicitly changing the 7702 EIP text to specify `tx.chain_id` as `U256`. Right now it only defines `auth.chain_id` to be `U256`

Reply:
febb39764 should now reflect the ethereum/EIPs#9143.